src/services/redisSessionStorage.ts

```diff
@@ -44,23 +55,69 @@ export default class RedisSessionStorage implements ISessionStorage {
         user.session_token,
         "EX",
         this.tokenDurationSecs,
-        (err, response) => resolve(this.singleStringReply(err, response))
+        (err, response) =>
+          resolve(
+            this.falsyResponseToError(
+              this.singleStringReply(err, response),
+              new Error("Error setting wallet token")
+            )
+          )
       );
     });
 
-    const [setSessionTokenResult, setWalletTokenResult] = await Promise.all([
+    const newSessionInfo: SessionInfo = {
+      createdAt: new Date(),
+      sessionToken: user.session_token
+    };
+    const sessionInfoKey = `${sessionInfoKeyPrefix}${user.session_token}`;
+    const saveSessionInfo = new Promise<Either<Error, boolean>>(resolve => {
+      this.redisClient.set(
+        sessionInfoKey,
+        JSON.stringify(newSessionInfo),
+        "EX",
+        this.tokenDurationSecs,
+        (err, response) =>
+          resolve(
+            this.falsyResponseToError(
+              this.singleStringReply(err, response),
+              new Error("Error setting user token info")
+            )
+          )
+      );
+    });
+    const updateSessionInfoSet = new Promise<Either<Error, boolean>>(
+      resolve => {
+        this.redisClient.sadd(
+          `${userSessionsSetKeyPrefix}${user.fiscal_code}`,
+          sessionInfoKey,
+          (err, response) =>
+            resolve(
+              this.falsyResponseToError(
+                this.integerReply(err, response),
+                new Error("Error updating uset tokens info set")
+              )
+            )
+        );
+      }
+    );
+    const setPromisesResult = await Promise.all([
       setSessionToken,
-      setWalletToken
+      setWalletToken,
+      saveSessionInfo,
+      updateSessionInfoSet
     ]);
-
-    if (isLeft(setSessionTokenResult) || isLeft(setWalletTokenResult)) {
-      return left<Error, boolean>(new Error("Error setting the token"));
-    }
-
-    if (!setSessionTokenResult.value || !setWalletTokenResult.value) {
-      return left<Error, boolean>(new Error("Error setting the token"));
+    const isSetFailed = setPromisesResult.some(isLeft);
+    if (isSetFailed) {
+      return left<Error, boolean>(
+        new Error(
+          setPromisesResult
+            .filter(isLeft)
+            .map(_ => _.value.message)
```

Comment:
this is the culprit of the weird logging string, I guess we'll have to create a new story for that

Reply:
https://www.pivotaltracker.com/story/show/166785497